ext/exd/provider/github/client.go

```diff
@@ -0,0 +1,123 @@
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"runtime"
+	"strings"
+
+	"github.com/odpf/optimus/ext/exd"
+)
+
+const provider = "github"
+
+// Client defines github client
+type Client struct {
+	ctx      context.Context //nolint:containedctx
+	httpdoer exd.HTTPDoer
```

Comment:
Can we use `httpClient  *http.Client` instead of `Doer`. This seems like a change for testing. With httpClient you can use httptest packages from standard library to test.

Example blogpost for using httptest: https://hassansin.github.io/Unit-Testing-http-client-in-Go
Please let me know if you want to discuss this.